### PR TITLE
(WIP) Fix surface refinement

### PR DESCRIPTION
**Pull request by inducer**
This adds a failing test for the ability to refine toruses and spheres without large node-vertex inconsistencies.

I'm guessing this might have to do with the work in #97. I don't yet fully understand what's going on.

cc @alexfikl

### CODE REVIEW

**Comment by alexfikl** [comment]
What's supposed to fail? It works locally.

File: test/test_refinement.py [hunk 1]
```diff
@@ -331,6 +330,30 @@ def test_conformity_of_uniform_mesh(refinement_rounds):
     assert is_boundary_tag_empty(mesh, BTAG_ALL)
 
 
+@pytest.parametrize("mesh_name", ["torus", "icosphere"])
```
Comment (alexfikl) [inline test/test_refinement.py:333]:
```suggestion
@pytest.mark.parametrize("mesh_name", ["torus", "icosphere"])
```

**Comment by inducer** [comment]
I get
```
Traceback (most recent call last):
  File "/home/andreas/src/meshmode/test/test_refinement.py", line 360, in <module>
    exec(sys.argv[1])
  File "<string>", line 1, in <module>
  File "/home/andreas/src/meshmode/test/test_refinement.py", line 349, in test_refine_surfaces
    refined_mesh = refine_uniformly(mesh, 1)
  File "/home/andreas/src/meshmode/meshmode/mesh/refinement/__init__.py", line 991, in refine_uniformly
    refiner.refine_uniformly()
  File "/home/andreas/src/meshmode/meshmode/mesh/refinement/no_adjacency.py", line 61, in refine_uniformly
    return self.refine(flags)
  File "/home/andreas/src/meshmode/meshmode/mesh/refinement/no_adjacency.py", line 227, in refine
    new_mesh = Mesh(new_vertices, new_el_groups, is_conforming=(
  File "/home/andreas/src/meshmode/meshmode/mesh/__init__.py", line 779, in __init__
    assert _test_node_vertex_consistency(
  File "/home/andreas/src/meshmode/meshmode/mesh/__init__.py", line 983, in _test_node_vertex_consistency
    assert _test_node_vertex_consistency_resampling(mesh, mgrp, tol)
  File "/home/andreas/src/meshmode/meshmode/mesh/__init__.py", line 971, in _test_node_vertex_consistency_resampling
    assert max_el_vertex_error < tol*size, max_el_vertex_error
AssertionError: 9.901371953343942
```
Strangely, that seems to reproduce on my laptop and nowhere else.

**Comment by alexfikl** [comment]
Hm, I've added a `refine_uniformly` to various surface tests in `test_meshmode.py` too and couldn't get any of them to fail (ran multiple times).

Does it fail for both the sphere and the torus?

**Comment by inducer** [comment]
> Does it fail for both the sphere and the torus?

Yeah. The fact that CI passed is further indication that this might me something to do with my environment. I'll need to investigate. Thanks for helping!

**Comment by inducer** [comment]
https://github.com/inducer/pytential/issues/63 is another thing that seems to be unique to my laptop. Kind of weird.

**Comment by inducer** [comment]
Very strangely, this works OK on my laptop in an environment based on conda.

**Comment by alexfikl** [comment]
> Very strangely, this works OK on my laptop in an environment based on conda.

I'm not using conda locally, just good old venv with master from pretty much everything.

**Comment by inducer** [comment]
On my laptop: Rebuilt my venv with the existing source trees... same result. Freshly downloaded source with fresh venv... also same result.

On dunkel: Freshly downloaded source with fresh venv... no error.

**Pull request merged**